Multisig/UI/Settings/SafeSettingsViewController/SafeSettingsViewController.swift

```diff
@@ -446,15 +447,17 @@ class SafeSettingsViewController: LoadableViewController, UITableViewDelegate, U
             show(editSafeNameViewController, sender: self)
 
         case Section.RequiredConfirmations.confirmations(_):
-            guard let isReadOnly = safe?.isReadOnly else {
+            guard let safe = safe,
+                  !safe.isReadOnly,
+                  safe.ownersInfo?.count ?? 1 >= 2,
+                  safeOwners.count >= 2 else {
```

Comment:
same here, the logic check should be moved somewhere else. Maybe to the ChangeConfirmationsFlow initializer and make the initializer nillable